app/controllers/reactions_controller.rb

```diff
@@ -53,7 +55,17 @@ def create
 
     Rails.cache.delete "count_for_reactable-#{params[:reactable_type]}-#{params[:reactable_id]}"
 
+    if params[:reactable_type] == "Article"
+      clear_article_reactions(
```

Comment:
I don't understand this part, why is the code removing all existing reactions?

Reply:
@rhymes, the idea behind this was to prevent the selection of both the `upVote` reaction and either of the `downVote` or `vomit` reactions.

In the view, the class `reacted` is added to a reaction dynamically if it has any cached reactions (see `app > views > moderations > actions_panel`):
`<button class="reaction-button <%= Reaction.cached_any_reactions_for?(@moderatable, current_user, "thumbsup") ? "reacted" : "" %>"></button>`

For the UI to be accurate, we needed to clear the reactions before adding the `reacted` class to appropriate buttons in the UI

Hope this makes sense?

Reply:
Got it! Thanks. What happens if we accidentally post `like` or `readinglist` as a category to this action?

Reply:
Great question! I will investigate.

Reply:
Sorry @msarit, I updated this! I added a check for the category type just to make the code clearer. That said, posting a `like` or `readinglist` wouldn't have done anything with the other reactions so I think we're okay here :)